pkg/test/resource_creator.go

```diff
@@ -48,8 +52,48 @@ func (ctx *Context) GetNamespace() (string, error) {
 	return ctx.namespace, nil
 }
 
+func (ctx *Context) GetOperatorNamespace() (string, error) {
+	if ctx.operatorNamespace != "" {
+		return ctx.operatorNamespace, nil
+	}
+	// create namespace
+	ctx.operatorNamespace = ctx.GetID()
+	namespaceObj := &core.Namespace{ObjectMeta: metav1.ObjectMeta{Name: ctx.operatorNamespace}}
+	_, err := ctx.kubeclient.CoreV1().Namespaces().Create(namespaceObj)
+	if apierrors.IsAlreadyExists(err) {
+		return "", fmt.Errorf("namespace %s already exists: %w", ctx.operatorNamespace, err)
+	} else if err != nil {
+		return "", err
+	}
+	ctx.AddCleanupFn(func() error {
+		return ctx.kubeclient.CoreV1().Namespaces().Delete(ctx.operatorNamespace, metav1.NewDeleteOptions(0))
+	})
+	return ctx.operatorNamespace, nil
+}
+
+func (ctx *Context) GetWatchNamespace() (string, error) {
```

Comment:
```suggestion
// GetWatchNamespace will return the  namespaces to operator watch for changes, if the flag --watch-namespaced not be used than it will  return the Operator Namespace. 
func (ctx *Context) GetWatchNamespace() (string, error) {
```

Reply:
done